File: src/main/g8/src/main/java/$package$/$agentName$.java
import org.enmas.pomdp.Agent;
import org.enmas.pomdp.Action;
import org.enmas.pomdp.State;

public class $agentName$ extends Agent {

	// See the API docs
	// http://enmas.github.com/api/enmas-core/#org.enmas.pomdp.Agent

	public String name() {
		return "$agentName$";
	}

	public Action policy(observation: State, reward: Float) {
		// inspect your observation
		// consider your reward
		// choose an action
	}

}

File: src/main/g8/src/main/scala/$package$/$agentName$.scala
import org.enmas.pomdp.{Agent, Action, State}
import scala.util.Random
 
class $agentName$ extends Agent {

  // See the API docs at
  // http://enmas.github.com/api/enmas-core/#org.enmas.pomdp.Agent

  def name = "$agentName$"

  def policy(observation: State, reward: Float): Action = {
    // inspect your observation
    // consider your reward
    // choose an action
  }

}
